homeassistant/config_entries.py

```diff
@@ -1561,6 +1557,29 @@ async def async_finish_flow(
                 self.config_entries.async_update_entry(
                     entry, discovery_keys=new_discovery_keys
                 )
+
+    async def async_flow_aborted(
+        self,
+        flow: data_entry_flow.FlowHandler[ConfigFlowContext, ConfigFlowResult],
+        result: ConfigFlowResult,
+    ) -> None:
+        """Handle an aborted config flow."""
+        self._cleanup_flow(cast(ConfigFlow, flow), result)
```

Comment:
Note that this is *only* called if a flow is already aborted after a step method is done.
It's not called if the step returns abort.

The name of the method should probably be adjusted to make this clearer.

Reply:
Maybe `async_flow_aborted_between_steps` or `async_flow_aborted_inflight`

Reply:
It could be the flow calling abort itself from within the step, or a race with the flow open in two browsers.

Reply:
No longer relevant after the approach is changed